runtime/scripts/part.js

```diff
@@ -1083,6 +1095,7 @@ if(res) { \
                 }
             }
         }
+        this.events.trigger('markAdaptive');
```

Comment:
I think this would be more useful to trigger at the top of the function, or maybe have "pre-" and "post-" events at either end.